client/src/pages/Report/CreateReport.tsx

```diff
@@ -0,0 +1,246 @@
+    const [generationType, setGenerationType] = useState<ReportType>('ON DEMAND');
+    const [generationPeriod, setGenerationPeriod] = useState<string>(schedulePeriodOptions.periods[2]);
+    const navigate = useNavigate();
+    const [startDateMs, setStartDateMs] = useState<number>(Date.now());
+    const [endDateMs, setEndDateMs] = useState<number>(Date.now());
+    const [showInfoPopup, setShowInfoPopup] = useState(false);
+
+    const handleDateRangeChange = (startMs: number, endMs: number) => {
+        setStartDateMs(startMs);
+        setEndDateMs(endMs);
+    };
+
+    useEffect(() => {
+        const fetchClusterDetails = async () => {
+            try {
+                const [clusterDetails, runningApplications, runningNodes] = await Promise.all([
+                    ManagmentServiceApiInstance.getClusterDetails(id || ''),
+                    ManagmentServiceApiInstance.getApplications(id || ''),
+                    ManagmentServiceApiInstance.getNodes(id || ''),
+                ]);
+
+                const runningApplicationsMap = runningApplications.reduce((acc, app) => {
+                    acc[`${app.name}-${app.kind}`] = app.running;
+                    return acc;
+                }, {} as Record<string, boolean>);
+
+                const runningNodesMap = runningNodes.reduce((acc, node) => {
+                    acc[node.name] = node.running;
+                    return acc;
+                }, {} as Record<string, boolean>);
+
+                const mappedNotificationChannels = [
+                    ...clusterDetails.slackReceivers.map(receiver => ({
+                        id: receiver.id.toString(),
+                        name: receiver.receiverName,
+                        details: receiver.webhookUrl,
+                        service: 'SLACK' as NotificationChannelKind,
+                        added: dateFromTimestampMs(receiver.createdAt),
+                        updated: dateFromTimestampMs(receiver.updatedAt),
+
+                    })),
+                    ...clusterDetails.discordReceivers.map(receiver => ({
+                        id: receiver.id.toString(),
+                        name: receiver.receiverName,
+                        details: receiver.webhookUrl,
+                        service: 'DISCORD' as NotificationChannelKind,
+                        added: dateFromTimestampMs(receiver.createdAt),
+                        updated: dateFromTimestampMs(receiver.updatedAt),
+                    })),
+                    ...clusterDetails.emailReceivers.map(receiver => ({
+                        id: receiver.id.toString(),
+                        name: receiver.receiverName,
+                        details: receiver.receiverEmail,
+                        service: 'EMAIL' as NotificationChannelKind,
+                        added: dateFromTimestampMs(receiver.createdAt),
+                        updated: dateFromTimestampMs(receiver.updatedAt),
+                    })),
```

Comment:
Consider refactor to make it less repetative (and possibly move as much logic outside of the component as possible